sustaingym/envs/building/stochastic_uncontrollable_generator.py

```diff
@@ -0,0 +1,222 @@
+                this_n = len(this_observation_block)
+
+                this_first_season = this_observation_block[:this_n//4]
+                this_summer = this_observation_block[this_n//4:3*this_n//4]
+                this_last_season = this_observation_block[3*this_n//4:]
+                this_winter = np.vstack((this_first_season, this_last_season))
+
+                if i == 0:
+                    self.summer_observations = this_summer
+                    self.winter_observations = this_winter
+                else:
+                    self.summer_observations = np.vstack((self.summer_observations, this_summer))
+                    self.winter_observations = np.vstack((self.winter_observations, this_winter))
+        else:
+            num_observations = observation_data.shape[0]
+            self.summer_observations = observation_data[num_observations//4:3*num_observations//4, :]
+            self.winter_observations = np.vstack((observation_data[:num_observations//4, :], observation_data[3*num_observations//4:, :]))
+
+        # (len_of_season x num_episodes, dim_of_obs_vector)
+        self.summer_observations = np.array(self.summer_observations)
+        self.winter_observations = np.array(self.winter_observations)
+
+        return self.summer_observations, self.winter_observations
+
+    def get_nearest_block_size(self, obs_length: int, block_size: int):
+        """
+        Finds nearest block size to perfectly divide the length of the observation data.
+
+        Args:
+            obs_length: The number of observation vectors in the data.
+            block_size: The desired block size.
+
+        Returns:
+            (one of) low, high: The nearest block size to perfectly divide the length
+                of the data.
+        """
+        low = block_size
+        high = block_size
+        while obs_length%low != 0 and obs_length%high != 0:
+            low -= 1
+            high += 1
+            if low <= 1:
+                low = 2
+        if obs_length%low == 0: return low
+        elif obs_length%high == 0: return high
+
+    def get_empirical_dist(self, 
+                           season: Optional[str] = None, 
+                           this_season_observations: Optional[np.array] = None, 
+                           block_size_on_split: int = 100) -> scipy.stats.rv_continuous:
+        """
+        Fits a multivariate normal distribution to each ambient feature.
+
+        Args:
+            season: The desired season. Can be `None` if this_season_observations is
+                not None. Otherwise, can be `summer` or `winter` if data has been generated
+                and split into seasons through split_observations_into_seasons.
+            this_season_observations: The observation data for this season. Can be
+                `None` if user has generated and split data.
+            block_size_on_split: Desired block size for each ambient feature vector.
```

Comment:
I think it makes more sense for the `block_size` not to get modified. If it doesn't perfectly divide the observation data, then just truncate the observation data.